src/versionbits.h

```diff
@@ -57,6 +57,7 @@ class AbstractThresholdConditionChecker {
     virtual bool Condition(const CBlockIndex* pindex, const Consensus::Params& params) const =0;
     virtual int64_t BeginTime(const Consensus::Params& params) const =0;
     virtual int64_t EndTime(const Consensus::Params& params) const =0;
+    virtual int MinActivationHeight(const Consensus::Params& params) const { return 0; }
```

Comment:
Is there a reason that this isn't pure virtual (when the other simple getter functions are)?

```suggestion
    virtual int MinActivationHeight(const Consensus::Params& params) const =0;
```

Reply:
None of the others have reasonable defaults, but min activation height does. Also, having min activation height default to zero means the condition checker maintains the same behaviour if left as default, which then allows the tests to be patched independently to test the new behaviours, and also means there's no change needed to the warning checker in validation.

Reply:
Noting that MinActivationHeight=0, even though it is a height, is easily compatible with any Signet MTP configuration activation parameters (it's just disabling the delay between locked-in and active, which is OK for a signet since it is managed by a signer anyways).